vms/platformvm/network/warp.go

```diff
@@ -0,0 +1,336 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package network
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"sync"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/network/p2p/acp118"
+	"github.com/ava-labs/avalanchego/proto/pb/platformvm"
+	"github.com/ava-labs/avalanchego/snow/engine/common"
+	"github.com/ava-labs/avalanchego/vms/platformvm/state"
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp/message"
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp/payload"
+)
+
+const (
+	ErrFailedToParseWarpAddressedCall = iota + 1
+	ErrWarpAddressedCallHasSourceAddress
+	ErrFailedToParseWarpAddressedCallPayload
+	ErrUnsupportedWarpAddressedCallPayloadType
+
+	ErrFailedToParseJustification
+	ErrConversionDoesNotExist
+	ErrMismatchedConversionID
+
+	ErrInvalidJustificationType
+	ErrFailedToParseSubnetID
+	ErrMismatchedValidationID
+	ErrValidationDoesNotExist
+	ErrValidationExists
+	ErrFailedToParseRegisterSubnetValidator
+	ErrValidationCouldBeRegistered
+
+	ErrImpossibleNonce
+	ErrWrongNonce
+	ErrWrongWeight
+)
+
+var _ acp118.Verifier = (*signatureRequestVerifier)(nil)
+
+type signatureRequestVerifier struct {
+	stateLock sync.Locker
```

Comment:
same comment - maybe create an interface with only the required methods, and then create a struct that locks only for those, instead of ah-hoc locking in all methods?